hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java

```diff
@@ -163,6 +163,8 @@
   private Predicate<LayoutVersionProto> layoutMatchCondition;
   private Predicate<LayoutVersionProto> layoutMisMatchCondition;
 
+  private final ConfigurationSource configuration;
```

Comment:
Don't add this field.  Pass the conf in the CapacityPipelineChoosePolicy constructor.

Reply:
Thank you. I'll modify it and update JIRA.